gradle-plugin-core/src/main/java/wtf/emulator/gmd/EwDeviceTestRunTaskAction.java

```diff
@@ -0,0 +1,338 @@
+    }));
+    workParams.getSideEffects().set(testRunInput.getSideEffects());
+    workParams.getTimeout().set(testRunInput.getTestTimeout());
+    workParams.getFileCacheEnabled().set(testRunInput.getFileCacheEnabled());
+    workParams.getFileCacheTtl().set(testRunInput.getFileCacheTtl());
+    workParams.getTestCacheEnabled().set(testRunInput.getTestCacheEnabled());
+    workParams.getNumFlakyTestAttempts().set(testRunInput.getNumFlakyTestAttempts());
+    workParams.getFlakyTestRepeatMode().set(testRunInput.getFlakyTestRepeatMode());
+    workParams.getDisplayName().set(testRunInput.getDisplayName().orElse(testRunData.getTestRunId()));
+    workParams.getDnsServers().set(testRunInput.getDnsServers().map(list -> list));
+    workParams.getEgressTunnel().set(testRunInput.getEgressTunnel());
+    workParams.getEgressLocalhostForwardIp().set(testRunInput.getEgressLocalhostForwardIp());
+    workParams.getScmUrl().set(testRunInput.getScmUrl());
+    workParams.getScmCommitHash().set(testRunInput.getScmCommitHash());
+    workParams.getScmRefName().set(testRunInput.getScmRefName());
+    workParams.getScmPrUrl().set(testRunInput.getScmPrUrl());
+    workParams.getIgnoreFailures().set(testRunInput.getIgnoreFailures());
+    workParams.getAsync().set(testRunInput.getAsync());
+    workParams.getPrintOutput().set(testRunInput.getPrintOutput());
+    workParams.getTestTargets().set(testRunInput.getTestTargets());
+    workParams.getProxyHost().set(testRunInput.getProxyHost());
+    workParams.getProxyPort().set(testRunInput.getProxyPort());
+    workParams.getProxyUser().set(testRunInput.getProxyUser());
+    workParams.getProxyPassword().set(testRunInput.getProxyPassword());
+
+    RegularFile outputFile = testRunInput.getIntermediatesOutputs().get().file(testRunData.getVariantName() + ".json");
+    workParams.getOutputFile().set(outputFile);
+    // TODO(tauno): using the project path here is technically incorrect, but it doesn't matter for sync cli invokes
+    workParams.getTaskPath().set(testRunData.getProjectPath());
+
+    return workParams;
+  }
+
+  private void convertResultsXml(@NotNull DeviceTestRunParameters<EwDeviceTestRunInput> deviceTestRunParameters) {
+    // AGP expects the test results to be with a specific name (TEST-*.xml) in the output directory.
+    // See /java/com/android/build/gradle/internal/test/report/TestReport.java;l=80
+    File outputsDir = deviceTestRunParameters.getTestRunData().getOutputDirectory().getAsFile();
+    if (outputsDir.exists()) {
+      File resultsXml = new File(outputsDir, "results.xml");
+      if (resultsXml.exists()) {
+        File renamedResultsXml = new File(outputsDir, "TEST-results.xml");
+        if (resultsXml.renameTo(renamedResultsXml)) {
+          // Extract required metadata from deviceTestRunParameters
+          String deviceName = deviceTestRunParameters.getDeviceInput().getDevice().get();
+          String flavor = deviceTestRunParameters.getTestRunData().getVariantName();
```

Comment:
`variantName` also contains buildtype, not only flavor. Or is the flavor here separate from AGP's product flavours?

Reply:
what does vanilla GMD do when provisioning a local emulator? If it also has the project path then we might want to replicate?